infrastructure/Taskfile.yml

```diff
@@ -589,6 +589,31 @@ tasks:
       - sh: "[ ! -z {{.PROJECT_NAME}} ]"
         msg: "Env variable PROJECT_NAME is not set or empty."
 
+    lagoon:project:update:
+      desc: Add a project to a lagoon remote
+      deps: [lagoon:cli:config]
+      cmds:
+        # - We assume that there will only be a single remote pr core, so we
+        #   hardcode --openshift (aka remote) to 1.
+        # - We deploy the main and develop branch
+        # - We configure the main branch to be the production environment,
+        #   This primarily means that develop will auto-idle
+        # - We bump the max allowed dev environments (default is 5)
+        - |
+            lagoon update project \
+            --gitUrl {{.GIT_URL}} \
+            --openshift 1 \
+            --productionEnvironment main \
+            --developmentEnvironmentsLimit 25 \
+            --branches "^(main|develop)$" \
+            --project {{.PROJECT_NAME}}
+        - task: lagoon:project:deploykey
+      preconditions:
+      - sh: "[ ! -z {{.GIT_URL}} ]"
+        msg: "Env variable GIT_URL is not set or empty."
+      - sh: "[ ! -z {{.PROJECT_NAME}} ]"
+        msg: "Env variable PROJECT_NAME is not set or empty."
```

Comment:
A "variable check" feature has been introduced in the latest version of `go-task`.
You should be able to use that instead:
https://taskfile.dev/usage/#ensuring-required-variables-are-set

Reply:
Cool! I might make that a task for later, as we can use that in a lot of places and it seems tangential to this PR?